apps/web/src/components/workflow/FlowEditor.tsx

```diff
@@ -159,19 +165,18 @@ export function FlowEditor({
         const oldNode = nodes[i + 1] || { position: { x: 0, y: 120 } };
         const newId = (step._id || step.id) as string;
 
-        await triggerErrors('steps');
+        await onStepInit?.(step);
 
         const newNode = buildNewNode(newId, oldNode, parentId, step, i);
-
-        const newEdge = buildNewEdge(parentId, newId);
-
         setNodes((nds) => nds.concat(newNode));
+
+        const newEdge = buildNewEdge(parentId, newId, edgeTypes ? 'special' : 'default');
```

Comment:
The condition of the third argument is a bit hidden and can be missed for someone not familiar with code. I'd suggest to extract it so it is a bit more clear.